### PR TITLE
Add test cases to consensus params validate

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Closes: #1379 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **Refactor**
	- Improved error handling in consensus parameter validation for enhanced clarity in error messages.
- **Tests**
	- Added new test cases for consensus parameter validation to ensure robustness and reliability.

### CODE REVIEW

File: types/params.go [hunk 1]
```diff
@@ -7,40 +7,68 @@ import (
...
 
+var (
+	// ErrMaxBytesCannotBeZero indicates that the block's MaxBytes cannot be 0.
+	ErrMaxBytesCannotBeZero = errors.New("block.MaxBytes cannot be 0")
+
+	// ErrMaxBytesInvalid indicates that the block's MaxBytes must be either -1 or greater than 0.
+	ErrMaxBytesInvalid = errors.New("block.MaxBytes must be -1 or greater than 0")
+
+	// ErrMaxBytesTooBig indicates that the block's MaxBytes exceeds the maximum allowed size.
+	ErrMaxBytesTooBig = errors.New("block.MaxBytes is too big")
```
Comment (MSevey) [inline types/params.go:18]: is there a way that we can provide direction here? either by stating the max size or saying how to find it?

File: types/params.go [hunk 2]
```diff
@@ -7,40 +7,68 @@ import (
...
+
+	// ErrMaxGasInvalid indicates that the block's MaxGas must be greater than or equal to -1.
+	ErrMaxGasInvalid = errors.New("block.MaxGas must be greater or equal to -1")
```
Comment (MSevey) [inline types/params.go:21]: really MaxGas can be zero? Wouldn't that prevent txns from being submitted? Or is it possible for gasless txns to be included if there isn't a lot of competition?

File: types/params.go [hunk 3]
```diff
@@ -7,40 +7,55 @@
...
+		return ErrPubKeyTypesEmpty
 	}
 
 	// Check if keyType is a known ABCIPubKeyType
 	for i := 0; i < len(params.Validator.PubKeyTypes); i++ {
 		keyType := params.Validator.PubKeyTypes[i]
 		if _, ok := cmtypes.ABCIPubKeyTypesToNames[keyType]; !ok {
-			return fmt.Errorf("params.Validator.PubKeyTypes[%d], %s, is an unknown pubkey type",
+			return fmt.Errorf("%w: params.Validator.PubKeyTypes[%d], %s,",
+				ErrPubKeyTypeUnknown,
```
[the remark this answers is not in the document]
Comment (MSevey) [reply types/params.go:71]: +1

**Review by MSevey — CHANGES_REQUESTED** [review]
couple minor edits, otherwise LGTM 👍

File: types/params.go (hunk 2 above)
Comment (Manav-Aggarwal) [reply types/params.go:21]: Yes, seems like it: https://docs.cometbft.com/v0.34/spec/abci/apps#blockparamsmaxgas, left for the app to enforce.

File: types/params.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply types/params.go]: Added a link to https://github.com/cometbft/cometbft/blob/2cd0d1a33cdb6a2c76e6e162d892624492c26290/types/params.go#L16 in the explaining comment.


Side note:

This is originally from here: https://github.com/cometbft/cometbft/blob/830c650e76a3049a9e1bb7b5e84416691bf6c187/types/params.go#L155

Interestingly, they only check for error vs no error case: https://github.com/cometbft/cometbft/blob/830c650e76a3049a9e1bb7b5e84416691bf6c187/types/params_test.go#L48

**Review by S1nus — APPROVED** [review]

**Review by MSevey — APPROVED** [review]

**Pull request merged**